packages/openapi3/src/schema-emitter.ts

```diff
@@ -936,12 +940,18 @@ export function getDefaultValue(program: Program, defaultType: Value): any {
       return null;
     case "EnumValue":
       return defaultType.value.value ?? defaultType.value.name;
-    default:
-      reportDiagnostic(program, {
-        code: "invalid-default",
-        format: { type: defaultType.valueKind },
-        target: defaultType,
-      });
+    case "ScalarValue":
+      if (modelProperty) {
+        return serializeValueAsJson(
+          program,
+          defaultType,
+          defaultType.type,
+          getEncode(program, modelProperty)
+        );
+      }
+      return serializeValueAsJson(program, defaultType, defaultType.type);
+    case "ObjectValue":
+      return serializeValueAsJson(program, defaultType, defaultType.type);
```

Comment:
```suggestion
      return serializeValueAsJson(program, defaultType, defaultType.type);
```

does this not work as the whole body of the function?

Reply:
it doesn't work for the encode part, to resolve the encode it needs the ModelProperty, so it can fall in this section: https://github.com/microsoft/typespec/blob/a336c9370940afd9a80173be9c31dab55d5597f6/packages/compiler/src/lib/examples.ts#L26

in the value is not resolved as a ModelProperty, the Scalar is not in the encode map, so is not correctly resolved.

I changed the code a little so it looks better and fits the method better :)